modules/loader/src/main/scala/com/snowplowanalytics/snowplow/postgres/config/ConfigPreCheck.scala

```diff
@@ -0,0 +1,30 @@
+package com.snowplowanalytics.snowplow.postgres.config
+
+import cats.effect.Sync
+import cats.data.EitherT
+import cats.implicits._
+
+import java.nio.file.{Path, Files}
+import com.snowplowanalytics.snowplow.postgres.config.LoaderConfig.Source.LocalFs
+
+import scala.util.Try
+
+/**
+  * Check if given configuration meets necessary requirements
+  */
+object ConfigPreCheck {
+  def apply[F[_]: Sync](cli: Cli[F]): EitherT[F, String, Cli[F]] =
+    EitherT(Sync[F].delay {
+      cli.config.input match {
+        // If configuration is for local source, check if given path exists
+        case LocalFs(path) =>
+          Either.fromTry(Try {
+            Files.exists(Path.of("/" + path.toString()))
```

Comment:
just a couple of naive questions

- why do we prepend `/`?
- what if user provides `~/...` or `/.../...` (any path that starts with `/`) ?

Reply:
I guess it would be good to give some background info about design choice of the library. `FileStore` is class responsible from local file operations. It is expecting file system root as argument. You can see example of this in [here](https://github.com/fs2-blobstore/fs2-blobstore#filestore) and related part in our project in [here](https://github.com/snowplow-incubator/snowplow-postgres-loader/pull/43/files#diff-6404f6977b418f97459551fb956ed00026be166387db29b2cc9dd46e69ba1aebR92). For example, if '/tmp' path is given as file system root to `FileStore` when we try to read '/snowplow/postgres' path it will read '/tmp/snowplow/postgres' path actually.

So for the first question, I prepend `/` because `toString` method is returning path without initial slash. If you pass '/snowplow/postgres' path, it will return 'snowplow/postgres'. I guess this is due to design choice I explained earlier.

For the second question, I didn't try paths with `~` or `./` to be honest. `~` maybe would work but I don't think current dir would work because I am giving `/` (root) as file system root hard-coded way currently. Maybe we can make also file system root configurable but I believe it would be complex to use and explain. Instead, we can advise user to use absolute path.

I guess it would be good to add some comments to explain this issue.

Reply:
Thanks for the explanation Enes, now I see how `FileStore` is used. If we will enforce absolute path usage, please make it clear in comment in config file and since any non-absolute path will be an invalid input, app should provide an explicit error message and quit, I think.

On the other hand, it feels like user should be able to provide any local path as s/he likes. I don't see why configuring fs root at runtime based on whether provided path is absolute or relative is complex to use and explain. If path is absolute, fs root is `/`, otherwise it is `.` / current directory. I also assume that if fs root is `.` , going upwards like `./../../input` is fine with the library.

Since this app supports Linux only, we can also handle special chars like `~` if we have the luxury.

Of course, all of this can be ignored or implemented in another release.

Reply:
When I said complicated, I was thinking making fsroot configurable but what you propose sounds better and it does not add any new config field. As you said, if given path is absolute fsroot will be `/`, otherwise it will be `.` Then I will change the implementation to this way. 👍 

Does it sound good to you also @chuwy ?